### PR TITLE
Distinguish URLs from paths in the treeinfo support

### PR DESCRIPTION
**Pull request by poncovka**
It is important to know if we are working with a URL or a path.

### CODE REVIEW

**Comment by poncovka** [comment]
Rebased and tested.

**Review by jkonecny12 — APPROVED** [review]
LGTM!

File: pyanaconda/modules/payloads/payload/dnf/tree_info.py [hunk 1]
```diff
@@ -400,38 +409,33 @@ def enabled(self):
...
-            return root_path
+            return root_url
+
+        # Get the protocol.
+        protocol, root_path = split_protocol(root_url)
 
         # Create the absolute path.
-        full_path = os.path.join(root_path, relative_path)
-        protocol, url = split_protocol(full_path)
+        absolute_path = os.path.join(root_path, relative_path)
```
Comment (VladimirSlavik) [inline pyanaconda/modules/payloads/payload/dnf/tree_info.py:437]: Did you want to use `pyanaconda.core.path.join_paths` here?

Comment (poncovka) [reply pyanaconda/modules/payloads/payload/dnf/tree_info.py:437]: Fixed in https://github.com/rhinstaller/anaconda/pull/4047.

**Pull request merged**